apps/arena/native/physics/src/collision_detection/sat.rs

```diff
@@ -77,16 +67,24 @@ pub(crate) fn intersect_circle_polygon(
 
         let min_depth = f32::min(circle_overlap_depth, polygon_overlap_depth);
 
+        if invalid_axis(
+            axis,
+            circle,
+            polygon,
+            &current_vertex,
+            &next_vertex,
+            obstacles,
+            external_wall,
+        ) {
+            continue;
+        }
```

Comment:
Why is this moved? Is the order of the code important?

Reply:
This was something i tried in the first iteration of this fix, since the issue was something else these lines should be reverted, i'll leave the `Position::mult` change since we're doing that for everything in out code base